src/Microsoft.AspNet.Mvc.Core/Resources.resx

```diff
@@ -451,4 +451,13 @@
   <data name="ModelType_WrongType" xml:space="preserve">
     <value>The model's runtime type '{0}' is not assignable to the type '{1}'.</value>
   </data>
+  <data name="TempData_CannotSerializeToSession" xml:space="preserve">
+    <value>The type {0} cannot be serialized to Session by '{1}'.</value>
+  </data>
+  <data name="TempData_CannotDeserializeToken" xml:space="preserve">
+    <value>Cannot deserialize JToken of type {0}.</value>
+  </data>
+  <data name="TempData_CannotSerializeDictionary" xml:space="preserve">
+    <value>The dictionary with TKey {0} cannot be serialized to Session by '{1}'.</value>
```

Comment:
`The '{0}' cannot serialize a dictionary with a key of type '{1}' to session state.` (Swapped `{0}` and `{1}` too.)